### PR TITLE
Update Eden Quest script and fix Issue#1173.

### PR DESCRIPTION
**Pull request by Emistry**
- Fix classes checking (https://github.com/HerculesWS/Hercules/issues/1173)
- Update and remove redundant script

---
This change is [Reviewable](https://reviewable.io/reviews/herculesws/hercules/1490)

### CODE REVIEW

File: npc/re/quests/eden/eden_quests.txt [hunk 1]
```diff
@@ -4079,21 +4072,10 @@ moc_para01,112,79,3	script	Weapons Expert BK#2nd11	4_M_REPAIR,{
...
-		"Eden Group Bow III:" +
-		"Eden Group Dictionary I:" +
-		"Eden Group Guitar I:" +
-		"Eden Group Whip I:" +
-		"Eden Group Revolver III:" +
-		"Eden Group Knuckles I:" +
-		"Eden Group Huuma Shuriken I";
+	.@wplist_size = getarraysize(.@wplist);
+	for ( .@i = 0; .@i <.@wplist_size; .@i++ ) {
+		.@menu$ = .@menu$ + getitemname(.@wplist[.@i]) + ":";
```
Comment (Asheraf) [inline npc/re/quests/eden/eden_quests.txt]: .@menu$ += getitemname(.@wplist[.@i]) + ":";

File: npc/re/quests/eden/eden_quests.txt [hunk 2]
```diff
@@ -1855,38 +1855,37 @@ moc_para01,112,96,5	script	Administrator Michael	4_M_RUSMAN1,{
...
 					P_Slayer1,"Eden Slayer I","Two-handed sword","Attack 162","",
 					P_Sabre1,"Eden Saber I","One-handed sword","Attack 147","This is what we strive for.";
 			}
-			if (Class == Job_Thief || Class == Job_Thief_High || Class == Job_Assassin || Class == Job_Rogue || Class == Job_Ninja || Class == Job_Novice || Class == Job_Novice_High) {
+			else if (BaseClass == Job_Thief || BaseClass == Job_Ninja || BaseClass == Job_Novice) {
 				callsub S_Select,"",
 					P_Dagger1,"Eden Dagger I","Dagger","MATK+60, attack 124","This is what we strive for.";
 			}
-			if (Class == Job_Merchant || Class == Job_Merchant_High || Class == Job_Blacksmith || Class == Job_Alchemist) {
+			else if (BaseClass == Job_Merchant || Class == Job_Merchant_High || Class == Job_Blacksmith || Class == Job_Alchemist) {
```
Comment (Lemongrass3110) [inline npc/re/quests/eden/eden_quests.txt]: Why not remove everything except the baseclass for this case?

File: npc/re/quests/eden/eden_quests.txt [hunk 3]
```diff
@@ -1933,34 +1932,34 @@ moc_para01,112,96,5	script	Administrator Michael	4_M_RUSMAN1,{
...
 			}
 			mes "[Michael]";
-			if (Class == Job_Swordman || Class == Job_Swordman_High || Class == Job_Knight || Class == Job_Crusader) {
+			if (BaseClass == Job_Swordman) {
 				callsub S_Select,"We have one and Two-handed swords",
 					P_Sabre2,"Eden Saber II","One-handed sword","Attack 170","",
 					P_Slayer2,"Eden Slayer II","Two-handed sword","Attack 185","";
 			}
-			if (Class == Job_Thief || Class == Job_Thief_High || Class == Job_Assassin || Class == Job_Rogue || Class == Job_Novice || Class == Job_Novice_High || Class == Job_SuperNovice || Class == Job_Soul_Linker || Class == Job_Ninja) {
+			else if (BaseClass == Job_Thief || BaseClass == Job_Novice || BaseClass == Job_Taekwon || BaseClass == Job_Ninja) {
```
Comment (Asheraf) [inline npc/re/quests/eden/eden_quests.txt]: `Class == Job_Novice`

File: npc/re/quests/eden/eden_quests.txt (hunk 2 above)
Comment (Emistry) [inline npc/re/quests/eden/eden_quests.txt]: probably mislook or my PR messed up due to some issue during rebase this morning. Will try update it soon.

File: npc/re/quests/eden/eden_quests.txt (hunk 3 above)
Comment (Asheraf) [inline npc/re/quests/eden/eden_quests.txt]: Yeah you're right.

**Comment by Asheraf** [comment]
Oh you're right :P

2016-10-26 12:26 GMT+01:00 Emistry Haoyan <notifications@github.com>:

**Comment by Emistry** [comment]
something went wrong , probably accidentally commit into master branch of mine. will recreate a new one.

**Pull request closed without merging**